modules/apps/transfer/v2/keeper/msg_server_test.go

```diff
@@ -0,0 +1,597 @@
+package keeper_test
+
+import (
+	"bytes"
+	"time"
+
+	sdkmath "cosmossdk.io/math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	transfertypes "github.com/cosmos/ibc-go/v9/modules/apps/transfer/types"
+	clienttypes "github.com/cosmos/ibc-go/v9/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v9/modules/core/04-channel/types"
+	channeltypesv2 "github.com/cosmos/ibc-go/v9/modules/core/04-channel/v2/types"
+	commitmenttypes "github.com/cosmos/ibc-go/v9/modules/core/23-commitment/types"
+	ibctesting "github.com/cosmos/ibc-go/v9/testing"
+	mockv2 "github.com/cosmos/ibc-go/v9/testing/mock/v2"
+)
+
+// TestMsgSendPacketTransfer tests the MsgSendPacket rpc handler for the transfer v2 application.
+func (suite *KeeperTestSuite) TestMsgSendPacketTransfer() {
+	var payload channeltypesv2.Payload
+	var path *ibctesting.Path
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expError error
+	}{
+		{
+			"success",
+			func() {},
+			nil,
+		},
+		{
+			"failure: send transfers disabled",
+			func() {
+				suite.chainA.GetSimApp().TransferKeeperV2.SetParams(suite.chainA.GetContext(),
+					transfertypes.Params{
+						SendEnabled: false,
+					},
+				)
+			},
+			transfertypes.ErrSendDisabled,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+
+			path = ibctesting.NewPath(suite.chainA, suite.chainB)
+			path.SetupV2()
+
+			tokens := []transfertypes.Token{
+				{
+					Denom: transfertypes.Denom{
+						Base:  sdk.DefaultBondDenom,
+						Trace: nil,
+					},
+					Amount: ibctesting.DefaultCoinAmount.String(),
+				},
+			}
+
+			ftpd := transfertypes.NewFungibleTokenPacketDataV2(tokens, suite.chainA.SenderAccount.GetAddress().String(), suite.chainB.SenderAccount.GetAddress().String(), "", transfertypes.ForwardingPacketData{})
+			bz := suite.chainA.Codec.MustMarshal(&ftpd)
+
+			timestamp := suite.chainA.GetTimeoutTimestampSecs()
+			// TODO: note, encoding field currently not respected in the implementation. encoding is determined by the version.
+			// ics20-v1 == json
+			// ics20-v2 == proto
+			payload = channeltypesv2.NewPayload(transfertypes.ModuleName, transfertypes.ModuleName, transfertypes.V2, "json", bz)
```

Comment:
Shouldn't this be "proto"?

Reply:
Yes it should! I actually encountered some errors due to this while working on using the encoding! Not worth fixing it now since I have already a commit (which might end up here on in a separate PR) that fixes it